src/Storages/ReadInOrderOptimizer.cpp

```diff
@@ -37,13 +37,13 @@ InputSortingInfoPtr ReadInOrderOptimizer::getInputOrder(const StoragePtr & stora
     {
         if (!merge_tree->hasSortingKey())
             return {};
-        sorting_key_columns = merge_tree->getSortingKeyColumns();
+        sorting_key_columns = merge_tree->getColumnsRequiredForSortingKey();
```

Comment:
But here we need `sorting_key_columns`, otherwise case of pk with function will be broken.